File: 프로그래머스/unrated/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    answer = []
    temp_arr = []

    for query in queries:
        for i in range(len(arr)):
            if query[0] <= i <= query[1] and arr[i] > query[2]:
                temp_arr.append(arr[i])
        
        if len(temp_arr) == 0:
            answer.append(-1)
        else:
            answer.append(min(temp_arr))
            temp_arr = []
            
    return answer

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.js
function solution(arr, queries) {
  let answer = [];

  for (let i = 0; i < queries.length; i++) {
    const result = arr
      .slice(queries[i][0], queries[i][1] + 1)
      .filter((v) => v > queries[i][2]);

    answer.push(result.length ? Math.min(...result) : -1);
  }

  return answer;
}